src/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py

```diff
@@ -0,0 +1,353 @@
+"""
+Vendored copy of runpy from the standard library.
+
+It's vendored so that we can properly ignore it when used to start user code
+while still making it possible for the user to debug runpy itself.
+
+runpy.py - locating and running Python code using the module namespace
+
+Provides support for locating and running Python scripts using the Python
+module namespace instead of the native filesystem.
+
+This allows Python code to play nicely with non-filesystem based PEP 302
+importers when locating support scripts as well as when importing modules.
+"""
+# Written by Nick Coghlan <ncoghlan at gmail.com>
+#    to implement PEP 338 (Executing Modules as Scripts)
+
+import sys
+import importlib.machinery  # importlib first so we can test #15386 via -m
+import importlib.util
+import io
+import types
+import os
+
+__all__ = [
+    "run_module", "run_path",
+]
+
+
+# Note: fabioz: Don't use pkgutil (when handling caught exceptions we could end up
+# showing exceptions in pkgutil.get_imported (specifically the KeyError), so,
+# create a copy of the function we need to properly ignore this exception when
+# running the program.
```

Comment:
What would be the consequence of using the one from stdlib? I'm curious because that's what debugpy does currently in launch scenarios.

Reply:
As a note, the patch changes debugpy to use this one instead of the one from the standard library too.

As far as debugpy is concerned, the main difference is that the stack related to runpy is no longer shown to the user (when justMyCode is false, which I believe is better for the user). Also, it was stopping inside of `runpy` when using `justMyCode` as false too along with `Raised Exceptions` multiple times...

For pydevd it had other repercussions because it treated the `_exec` inside of `pydevd.py` as the root, so, there were other issues related to unhandled exceptions which didn't work well (and a bunch of failing tests which didn't expect the changes in the stack) just by using `runpy` from the standard library with the pydevd entry point (it could also be fixed in pydevd. but it'd be more work than vendoring runpy and making that a part of the debugger).